src/k8s/pkg/k8sd/api/cluster_bootstrap.go

```diff
@@ -31,8 +31,12 @@ func (e *Endpoints) postClusterBootstrap(_ state.State, r *http.Request) respons
 	}
 
 	// Check if the cluster is already bootstrapped
-	_, err = e.provider.MicroCluster().Status(r.Context())
-	if err == nil {
+	status, err := e.provider.MicroCluster().Status(r.Context())
```

Comment:
We had a bug in main here: https://github.com/canonical/k8s-snap/blob/45db693e456d00598833502b726389eb906813fc/src/k8s/pkg/k8sd/api/cluster_bootstrap.go#L34
That version uses microcluster v3. This version contained a [bug that was fixed in a later version](https://github.com/canonical/microcluster/pull/264) 
We silenced that issue here: https://github.com/canonical/k8s-snap/blob/0e4cd5c39c7df62a270450a0323539806881d593/src/k8s/pkg/k8sd/api/cluster_bootstrap.go#L33-L34
Instead of checking for the error we need to check for status.ready from that function.

Interestingly, this issue popped now up in v2 because the bugfix was backported with https://github.com/canonical/microcluster/pull/265/files and therefore no error was thrown anymore, causing the cluster falsely to claim that it is already bootstrapped.

Reply:
Great find @bschimke95! Thanks for resolving this bug.